shell/platform/darwin/common/framework/Source/FlutterStandardCodec.mm

```diff
@@ -364,16 +366,22 @@ - (UInt32)readSize {
   }
 }
 
-- (NSData*)readData:(NSUInteger)length {
+- (CFDataRef)readDataRef:(NSUInteger)length {
   _range.length = length;
   NSData* data = [_data subdataWithRange:_range];
   _range.location += _range.length;
-  return data;
+  // return (CFDataRef)CFAutorelease((__bridge CFDataRef)data);
+  return (CFDataRef)CFRetain((__bridge CFDataRef)data);
+}
+
+- (NSData*)readData:(NSUInteger)length {
+  return (NSData *)CFBridgingRelease([self readDataRef:length]);
 }
 
 - (NSString*)readUTF8 {
-  NSData* bytes = [self readData:[self readSize]];
-  return [[NSString alloc] initWithData:bytes encoding:NSUTF8StringEncoding];
+  CFDataRef bytes = (CFDataRef)[self readDataRef:[self readSize]];
+  NSData *data = (NSData*)CFBridgingRelease(bytes);
```

Comment:
Why do you do `CFBridgingRelease` here instead of just using `(__bridge)`?  `bytes` is already on the autorelease pool and `initWithData:encoding:` should retain it.  There's no need to involve ARC.  That's what we are trying to avoid =)